src/main/java/com/datadoghq/connect/logs/util/RetryUtil.java

```diff
@@ -0,0 +1,57 @@
+package com.datadoghq.connect.logs.util;
+
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Utility to compute the retry times for a given attempt, using exponential backoff.
+ * <p>
+ * The purposes of using exponential backoff is to give the Datadog service time to recover when it becomes overwhelmed.
+ * Adding jitter attempts to prevent a thundering herd, where large numbers of requests from many tasks overwhelm the
+ * Datadog service, and without randomization all tasks retry at the same time. Randomization should spread the retries
+ * out and should reduce the overall time required to complete all attempts.
+ */
+public class RetryUtil {
+
+    /**
+     * An arbitrary absolute maximum practical retry time.
+     */
+    public static final long MAX_RETRY_TIME_MS = TimeUnit.HOURS.toMillis(24);
+
+    /**
+     * Compute the time to sleep using exponential backoff with jitter. This method computes the normal exponential backoff
+     * as {@code initialRetryBackoffMs << retryAttempt}, and then chooses a random value between 0 and that value.
+     *
+     * @param retryAttempts the number of previous retry attempts; must be non-negative
+     * @param initialRetryBackoffMs the initial time to wait before retrying; assumed to be 0 if value is negative
+     * @return the non-negative time in milliseconds to wait before the next retry attempt, or 0 if {@code initialRetryBackoffMs} is negative
+     */
+    public static long computeRandomRetryWaitTimeInMillis(int retryAttempts, long initialRetryBackoffMs) {
```

Comment:
Note: Instead of using a static function, you can use an instance of a class. It makes the usage easier especially if the logic is called from several locations. It also makes unit test easier by creating an interface and mock it in the unit test.
```java
class ExponentialBackOff
{
      void reset()
      long getNextWaitTimeMs()
}
```